Mathlib/CategoryTheory/Shift/ShiftSequence.lean

```diff
@@ -191,6 +193,68 @@ lemma shiftIso_hom_app_comp (n m mn : M) (hnm : m + n = mn)
   rw [F.shiftIso_add'_hom_app n m mn hnm a a' a'' ha' ha'', ← Functor.map_comp_assoc,
     Iso.inv_hom_id_app, Functor.map_id, id_comp]
 
+/-- The morphism `(F.shift a).obj X ⟶ (F.shift a').obj Y` induced by a morphism
+`f : X ⟶ Y⟦n⟧` when `n + a = a'`. -/
+def shiftMap {X Y : C} {n : M} (f : X ⟶ Y⟦n⟧) (a a' : M) (ha' : n + a = a') :
+    (F.shift a).obj X ⟶ (F.shift a').obj Y :=
+  (F.shift a).map f ≫ (F.shiftIso _ _ _ ha').hom.app Y
+
+@[reassoc]
+lemma shiftMap_comp {X Y Z : C} {n : M} (f : X ⟶ Y⟦n⟧) (g : Y ⟶ Z) (a a' : M) (ha' : n + a = a') :
+    F.shiftMap (f ≫ g⟦n⟧') a a' ha' = F.shiftMap f a a' ha' ≫ (F.shift a').map g := by
+  simp [shiftMap]
+
+@[reassoc]
+lemma shiftMap_comp' {X Y Z : C} {n : M} (f : X ⟶ Y) (g : Y ⟶ Z⟦n⟧) (a a' : M) (ha' : n + a = a') :
+    F.shiftMap (f ≫ g) a a' ha' = (F.shift a).map f ≫ F.shiftMap g a a' ha' := by
+  simp [shiftMap]
+
+lemma shiftIso_hom_app_comp_shiftMap {X Y : C} {m : M} (f : X ⟶ Y⟦m⟧) (n mn : M) (hnm : m + n = mn)
```

Comment:
I'll leave this as optional but personally I think it might be nice to have some kind of ASCII art for this lemma and the next one just showing what they actually *say*, because it's very hard to read the statements, even within VS Code.

Reply:
Thanks for the suggestion. I have added docstrings with how I understand what these lemmas say.